nearlib/signing/account_info.js

```diff
@@ -0,0 +1,79 @@
+const KeyPair = require('./key_pair');
+
+/**
+ * Standard format for storing account info in json
+ */
+class AccountInfo {
+
+    constructor(accountId, keyPair, networkId) {
+        this.accountId = accountId;
+        this.keyPair = keyPair;
+        this.networkId = networkId;
+    }
+
+    /**
+     * Reconstruct account info object from json.
+     * @param {Object} json 
+     */
+    static fromJson(json) {
+        if (!json.public_key || !json.secret_key || !json.account_id || !json.network_id) {
+            throw 'Invalid account info format. Please ensure it contains public_key, secret_key, and account_id".';
+        }
+        return new AccountInfo(json.account_id, new KeyPair(json.public_key, json.secret_key), json.network_id);
+    }
+
+    /**
+     * Convert to standard json format.
+     */
+    toJson() {
+        return {
+            account_id: this.accountId,
+            public_key: this.keyPair.getPublicKey(),
+            secret_key: this.keyPair.getSecretKey(),
+            network_id: this.networkId
+        };
+    }
+
+    /**
+     * Gets a key pair from account info.
+     */
+    getKeyPair() {
```

Comment:
getXXX methods aren't needed in JS, there is a way to override property getter in case you need other logic than returning field value:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get

Reply:
nit This is not a very good example because it defined a read only property generated from other data members. Simply changing this to get keyPair doesn't work because it now wants a setter. Adding a set keyPair doesn't work because the code in constructor is now interpreted as calling setter and creates a circular dependency. Now I have to rename the data fields to be shortened which is suboptimal.

Reply:
This method just returns `this.keyPair`. You can literally just remove the method in this use case. Same with all other getters except `getKeyFileName` – that one could have been replaced by `get keyFileName`.

If in some point in future accessing `keyPair` field directly wouldn't be sufficient – you'll be able to add `get keyPair` with whatever logic necessary.

Reply:
@janedegtiareva why did you add get/set handlers that do nothing?